### ISSUE TITLE
Add to Windows 10 Settings Default Applications

### ISSUE THREAD

**Issue by lainz** [bgrabitmap/lazpaint#97]
Hi, I think I need to setup LazPaint as default image viewer, since the Windows 10 image viewer is slow on my PC.

There is a setting on Windows 10:
![windows_default_apps](https://user-images.githubusercontent.com/4131395/69502467-6ce25900-0eee-11ea-8776-c3a6eb448310.png)

But LazPaint is not listed there.

I think this can be done using registry, but is a guess... maybe we can ask how Gimp did get added there in some Gimp forum?

I know that the default file associations can be done when we install LazPaint as well, but this Windows setting will be nice to have too.

**Comment by circular17** [comment]
There is probably some registry setting. Maybe this could be done by the installer?

**Comment by lainz** [comment]
Yes if it's a registry setting for sure, inno setup can handle registry.

**Comment by lainz** [comment]
Well Gimp registers more stuff, with this tool searching for "gimp" it shows more registry entries that lazpaint
https://www.nirsoft.net/utils/regscanner.html

But yet I need to test if adding these to lazpaint shows it in the list or not

**Comment by circular17** [comment]
Hmmm ok.

There is no documentation online about it ?

**Comment by lainz** [comment]
Not one I found that works for images, I found the gimp source code of installer but not have time to check it yet

https://gitlab.gnome.org/GNOME/gimp/blob/fe35294ee7c244367dfb4963b7c72c86d0e3ac44/build/windows/installer/associations.isi

**Comment by lainz** [comment]
https://stackoverflow.com/questions/32354861/how-to-find-the-default-browser-via-the-registry-on-windows-10

**Comment by lainz** [comment]
On the registry I found nothing, I compared VLC that's the video player that is listed on default video player and Gimp that is listed on image viewer, nothing special about these, just that these supports a lot of file extensions.

Maybe Windows detects the file associations each application supports (in capabilities / file associations, lazpaint is registered there) and if it match the file associations for the category it adds it automatically.

For example this app is listed, and these are the associations checked when I select it from the menu:
![windows_image_viewer](https://user-images.githubusercontent.com/4131395/69637626-c99d5b00-1037-11ea-92f0-765ad7cc7c0f.PNG)

Also there's nothing special in the gimp inno setup script.
